### DISCUSSION TITLE
QChatMessage default slot behavior for array of messages

### DISCUSSION THREAD

**Discussion by lyle45** [quasarframework/quasar#9142] [General - Components / Directives / etc] [answered]
Hi,

I couldn't find in the docs an example of how to make chat component default slot behave the same way it does when used as text prop. text prop can take an array of messages and display then in conjunction. How can this effect be achieved using slots? if default slot is treated as array (v-for on root element) then it just stuffs them together into the same message.
This can be necessary for multiple reasons like markdown parsing etc

e.g. 

``` vue
      <q-chat-message
        name="me"
        avatar="https://cdn.quasar.dev/img/avatar3.jpg"
        stamp="7 minutes ago"
        sent
        bg-color="amber-7"
     >
       <q-markdown :src="someMarkdown"/>
     </q-chat-message>
```

Taken from 
https://forum.quasar-framework.org/topic/4341/markdown-and-chat/6
where no answer is given exactly

example case here

https://codepen.io/lyle45/pen/ZELgJqB

**Comment by metalsadman** [comment]
@lyle45 
> where no answer is given exactly

what do you mean no answer? when you quoted the answer from the forum question. at least when asking, show us what you have tried first.

v-for'ing a q-chat is no different from v-for'ing other vue components, here using that snippet answered in that forum plus example from [docs](https://quasar.dev/vue-components/chat#example--default-slot) and array of objects for messages.

https://codepen.io/metalsadman/pen/BaLLOvM?editors=1010.

**Reply by lyle45** [reply]
@metalsadman 
I mean that below said question OP further mentioned that text is an array of messages. If I pass the array to the text prop I will get multiple message "balloons"  as if someone sent multiple messages one after the other

In you example all markdown array items are pushed into one message balloon. how to get same behavior as text prop using default slot?
You can achieve this behavior using only css but it's encapsulated inside the message. possible using negative margin, but hackish, is there a better way?

**Reply by lyle45** [reply]
@metalsadman v-for'ing inside default slot is like writing a bigger div into one message instead of displaying multiple associated messages

**Reply by lyle45** [reply]
@metalsadman If current behavior isn't  supported (which I assume it can't if just using the default slot), then it would be pretty easy supporting it be creating a new slot scope for #text slot and have it receive one text message from the array passed through the text prop do whatever you want with. probably should pass entire message object aswell for higher customizability

**Reply by metalsadman** [reply]
@lyle45 what you mean not supported? perhaps make a codepen of what exactly you have tried. you do see in my codepen i've used the text in the slot `{{text}}` it even has many stuff inside it that are "customized", it uses v-for, and etc... 

though if you really feel like it's lacking and should serve what your use-case is, then file a PR or a feature request.

**Reply by lyle45** [reply]
@metalsadman 

Based on your codepen, 
https://codepen.io/lyle45/pen/ZELgJqB
See the 2 differences. Again the white strip css can be achieved with negative margin so maybe this isn't that important, but it feels that slots should be able to mimic props functionality completely

opened feature request

https://github.com/quasarframework/quasar/issues/9148

**Reply by disfated** [reply]
@lyle45 the only solution right now is to pre-parse your text array with markdown and pass to `QChatMessage` along with `text-html` prop.
Something like https://codepen.io/disfated/pen/eYgqGxj

**Reply by lyle45** [reply]
> @lyle45 the only solution right now is to pre-parse your text array with markdown and pass to `QChatMessage` along with `text-html` prop.
> Something like https://codepen.io/disfated/pen/eYgqGxj

@disfated Thanks for solution but that's won't work for me. need to integrate markdown messages with image previews or video or other customizations so have to use slots

**Comment by lyle45** [comment]
Does anyone has any idea how to make this work normally?
really sticking me atm

**Comment by rstoenescu** [comment, marked as answer]
The required behavior will be available in Quasar v2.0.0-beta.16 and Quasar v1.15.14

**Reply by rstoenescu** [reply]
As per https://github.com/quasarframework/quasar/issues/9148
